### PR TITLE
feat: support generator functions in expressions

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR ports the generator functions to be executed as part of an expression.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: server/expression/functions/function.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		if arg.Type != expectedArgType {
+			return types.TypedValue{}, fmt.Errorf("invalid argument type on index %d: expected %s, got %s", i, arg.Type.String(), expectedArgType.String())
+		}
+	}
+
+	returnedValue := f.invoker(args...)
+	return types.GetTypedValue(returnedValue), nil
+}
+
+func GetFunctionRegistry() Registry {
```
Comment (schoren) [inline server/expression/functions/function.go:33]:
```suggestion
func DefaultRegistry() Registry {
```

File: server/expression/parser.go [hunk 2]
```diff
@@ -31,6 +32,9 @@ func ParseStatement(statement string) (Statement, error) {
 		return Statement{}, fmt.Errorf("could not create parser: %w", err)
 	}
 
+	tokens, _ := parser.Lex("", bytes.NewBuffer([]byte(statement)))
+	fmt.Println(tokens)
+
```
Comment (schoren) [inline server/expression/parser.go:37]:
```suggestion
```

File: server/expression/filters/length.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/kubeshop/tracetest/server/expression/types"
+)
+
+func Length(input Value, args ...string) (Value, error) {
+	if len(args) != 0 {
+		return Value{}, fmt.Errorf("wrong number of args. Expected 0, got %d", len(args))
+	}
+
+	if len(input) != 1 {
+		return Value{}, fmt.Errorf("wrong input: Expected string, not an array")
```
Comment (schoren) [inline server/expression/filters/length.go:15]: could we make it work with arrays too? like `["1", "2"] | length = 2`

**Review by schoren — APPROVED** [review]
Looks awesome! Left a few suggestions, mainly would be awesome to have `lenght` work with arrays, but feel free to merge

File: docs/test-definition-file.md [hunk 4]
```diff
@@ -141,7 +141,7 @@ trigger:
 
 Sometimes we want to randomize our test data. Maybe we want to try new values or maybe we know our API will fail if the same id is provided more than once. For this use case, you can define generator functions in the test trigger.
 
-To use a generator function, wrap it in double curly brackets: `{{ uuid() }}`.
+Generator functions can be invoked as part of expressions. Therefore, you only need to invoke it as `uuid()`. However, you might want to generate values and concatenate them with static texts as well. For this, you can use the string interpolation feature: `"your user id is ${uuid()}`
```
Comment (jfermi) [inline docs/test-definition-file.md:144]:
```suggestion
Generator functions can be invoked as part of expressions. Therefore, you only need to invoke it as `uuid()`. However, you might want to generate values and concatenate them with static texts as well. For this, you can use the string interpolation feature: `"your user id is ${uuid()}`.
```

File: docs/test-definition-file.md [hunk 5]
```diff
@@ -289,7 +289,7 @@ Using the same hypotethical user creation endpoint, a user creation might result
 - `UPDATE accounts SET remaining_users ...`
 
 In this case, the service is instrumented so that each query generates a span of type `database`.
-You can get a list of sql operations 
+You can get a list of sql operations
```
Comment (jfermi) [inline docs/test-definition-file.md:292]:
```suggestion
You can get a list of sql operations:
```

**Review by jfermi — COMMENTED** [review]
A few editing suggestions.

File: server/expression/filters/length.go (hunk 3 above)
Comment (mathnogueira) [reply server/expression/filters/length.go]: Done. Merged `count` and `length` together. Now `length` works with arrays and strings.

**Pull request merged**